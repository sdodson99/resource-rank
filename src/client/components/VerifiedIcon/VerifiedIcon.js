import React from 'react';
import PropTypes from 'prop-types';
import styles from './VerifiedIcon.module.css';
import { CheckCircle } from '@material-ui/icons';

const VerifiedIcon = ({ size }) => (
  <div className={styles.VerifiedIcon} data-testid="VerifiedIcon">
    <CheckCircle htmlColor={'#6f99f2'} style={{ fontSize: size }} />
  </div>
);

VerifiedIcon.propTypes = {
  size: PropTypes.number,
};

VerifiedIcon.defaultProps = {
  size: 20,
};

export default VerifiedIcon;
