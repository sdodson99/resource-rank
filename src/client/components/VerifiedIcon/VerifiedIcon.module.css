.VerifiedIcon {
  @apply flex items-center;
}
