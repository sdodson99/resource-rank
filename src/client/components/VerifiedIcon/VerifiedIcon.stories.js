import React from 'react';
import VerifiedIcon from './VerifiedIcon';

export default {
  component: VerifiedIcon,
  title: 'Components/VerifiedIcon',
};

const Template = (args) => <VerifiedIcon {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Size = Template.bind({});
Size.args = {
  size: 50,
};
