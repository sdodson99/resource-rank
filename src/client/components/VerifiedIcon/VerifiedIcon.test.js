import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import VerifiedIcon from './VerifiedIcon';

describe('<VerifiedIcon />', () => {
  it('should mount', () => {
    render(<VerifiedIcon />);

    const verifiedIcon = screen.getByTestId('VerifiedIcon');

    expect(verifiedIcon).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<VerifiedIcon />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with custom size', () => {
    const component = createRenderer().render(<VerifiedIcon size={100} />);

    expect(component).toMatchSnapshot();
  });
});
