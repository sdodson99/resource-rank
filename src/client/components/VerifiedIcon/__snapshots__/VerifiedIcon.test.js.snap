// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifiedIcon /> should render correctly 1`] = `
<div
  className="VerifiedIcon"
  data-testid="VerifiedIcon"
>
  <Memo
    htmlColor="#6f99f2"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</div>
`;

exports[`<VerifiedIcon /> should render correctly with custom size 1`] = `
<div
  className="VerifiedIcon"
  data-testid="VerifiedIcon"
>
  <Memo
    htmlColor="#6f99f2"
    style={
      Object {
        "fontSize": 100,
      }
    }
  />
</div>
`;
